### PR TITLE
[101] Implement edit rejected actor data UI

### PR DESCRIPTION
**Pull request by PKhing**
## What did you do

- implement edit rejected actor data UI
- please forgive me for a lot of TODO and any (will fix after backend implement the API)
- fix all guards to cover REJECTED case
- already responsive
image

## Demo

- login with `actor10@gmail.com` should redirect to /rejected page

### CODE REVIEW

File: apps/web/src/modules/rejected/pages/hooks/useInitialData/index.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+          reason: 'Initial Data Casting',
+          data: {},
+        })
+      else
+        setInitialData({
+          reason: 'Initial Data Actor',
+          data: {},
+        })
+    }, 1000)
+  }, [type])
```
Comment (miello) [inline apps/web/src/modules/rejected/pages/hooks/useInitialData/index.ts:24]: (Question)
this is temporary, right ?

File: apps/web/src/modules/rejected/actor/types.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export interface EditActorInfoFormProps {
+  initialData: any
```
Comment (miello) [inline apps/web/src/modules/rejected/actor/types.ts:2]: any 🤔

**Review by miello — CHANGES_REQUESTED** [review]
(This will include only code checking)

I think we have too many guards with repetitive functionality now
```
In summary

user is null => / or /login
status is ACCEPTED => /job
status is PENDING => /waiting
status is REJECTED => /rejected
```

Can we refactor it to 1 guard with some options for making it more compact and centralized but it will increase coupling to HOC guard instead

File: apps/web/src/modules/rejected/pages/hooks/useInitialData/index.ts (hunk 1 above)
Comment (PKhing) [reply apps/web/src/modules/rejected/pages/hooks/useInitialData/index.ts:24]: Yep

File: apps/web/src/modules/rejected/actor/types.ts (hunk 2 above)
Comment (PKhing) [reply apps/web/src/modules/rejected/actor/types.ts:2]: image
There is no DTO :ngae:

**Comment by miello** [comment]
I think "reason" text is not quite attractive to user
image

**Comment by PKhing** [comment]
> I think "reason" text is not quite attractive to user image

Tell design not me 😂

**Review by miello — APPROVED** [review]

**Pull request merged**